src/backend/xdg_desktop_portal.rs

```diff
@@ -0,0 +1,284 @@
+use std::path::PathBuf;
+
+use crate::backend::DialogFutureType;
+use crate::file_dialog::Filter;
+use crate::{FileDialog, FileHandle};
+
+use ashpd::desktop::file_chooser::{
+    FileChooserProxy, FileFilter, OpenFileOptions, SaveFileOptions,
+};
+// TODO: convert raw_window_handle::RawWindowHandle to ashpd::WindowIdentifier
+use ashpd::{zbus, WindowIdentifier};
+
+use log::warn;
+use smol::block_on;
```

Comment:
if we are only using block_on then there is no reason to include full blown async executor like smol, I would rather just use `pollster` or `futures`